### ISSUE TITLE
MultiMatch "MostFields" Type conversion issue

### ISSUE THREAD

**Issue by RWAust** [elastic/elasticsearch-net#1428]
Hi

In one of my queries, I have include the following:

           q.MultiMatch(m2 => m2
                                    .Type(TextQueryType.MostFields)
                                    .Query(autoCompleteSearchTerm)
                                    .OnFields(t => t.Suffix("*.en"))
                                    .Analyzer("english")
                                    )

On my development, the query that includes the above code works perfectly and there are has never been an issue with it.

Now, when I try to deploy my project to the web server, which is a different machine to my development machine, the query that uses the code extract above now fails. Relevant log extracts from the elasticsearch log file report the following:

         "All shards failed for phase: [query] org.elasticsearch.search.SearchParseException"

           "multi_match": {
            "type": "MostFields",
            "query": "chi",
            "analyzer": "english",
            "fields": [
              "*.en"
            ]
          }

          Caused by: org.elasticsearch.ElasticsearchParseException: No type found for value: MostFields

 From the Nest enum TextQueryType, it would seem that what is supposed to happen is that the 

         .Type(TextQueryType.MostFields)

should get converted to

         "most_fields"

before the query is executed by elasticsearch. However, instead of receiving "most_fields", "MostFields" is received instead.

I currently have version 1.5.1 of both elasticsearch, Nest and elasticsearch.Net installed on both the development machine (Windows 7) and the server machine Windows Server 2012.

I have trawled through the server to ensure that there is no older Nest.dll or Elasticsearch.Net.dll file being called.

What is not clear is why am I am having this "type" issue on the server machine and not the development machine where elasticsearch, Nest.dll and Elasticsearch.Net.dll appears to be identical on both machines?

Thanks in advance for your help.

**Comment by RWAust** [comment]
To be sure that problem is not related to the IIS7 server, I have completely uninstalled and then re-installed the web server. Unfortunately, the problem still exists where the elastic instance is incorrectly receiving "MostFields" instead of "most_fields" in the MultiMatch 'type' field.

Any help will be greatly appreciated.

**Comment by gmarz** [comment]
Hey @RWAust

Hmm...Json.NET's `StringEnumConverter` is responsible for serializing the enum.  It looks like it's doing its job by serializing to `MostFields` rather than the integer value, however, for some reason it doesn't look like it's taking into account the [EnumMember](https://github.com/elastic/elasticsearch-net/blob/develop/src/Nest/Enums/TextQueryType.cs#L12)s that are decorated on the values.

Can you confirm the version of `Newtonsoft.Json.dll`, the target framework version of your app?

**Comment by RWAust** [comment]
Hi gmarz

The version of Newonsoft.Json.dll is 6.0.1.17001
Target framework version is 4.5

**Comment by RWAust** [comment]
Hi gmarz

I upgraded the Newonsoft.Json.dll to the latest version of 6.0.8.18111, but unfortunately this did not resolve the problem.

RW

**Comment by gmarz** [comment]
Hmm...this is a head scratcher.

Either of those Newtonsoft.Json versions should be fine.

Here's the bit in Json.Net that reads the `EnumMemberAttribute`: https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/Utilities/EnumUtils.cs#L54

What versions of .NET are installed on the server?

**Comment by RWAust** [comment]
Hi gmarz

It sure is a head scratcher. As far as I could tell, all the relevant files on the server matched the files on the development machine, and despite trying many different things, I could not resolve this problem. 

In the end, I decided to take the drastic action and rebuild the entire server. A day later, I was able to test for the problem and now my query works as expected; "MostFields" is now passed in as "most_fields".

Thanks gmarz for your assistance.

RW.
